common/chat/contacts.ts

```diff
@@ -0,0 +1,141 @@
+import assert from 'assert';
+// import { Pupil, Student } from '../../graphql/generated';
+import { prisma } from '../prisma';
+import { isPupil, isStudent, userForPupil, userForStudent } from '../user';
+import { pupil as Pupil, student as Student } from '@prisma/client';
+import { User } from '../user';
+
+export type UserContactType = {
+    userID: string;
+    firstname: string;
+    lastname: string;
+    email: string;
+};
+
+type UserContactlist = {
+    [userId: string]: {
+        user: UserContactType;
+        contactReasons: string[];
+    };
+};
+
+const getMatchContacts = async (user: User): Promise<Student[] | Pupil[]> => {
+    if (user.pupilId) {
+        return await prisma.student.findMany({
+            where: {
+                match: { some: { pupilId: user.pupilId } },
```

Comment:
@Jonasdoubleyou can I get the match id back here?

Reply:
No because one student can have multiple matches (though not with the same pupil)